core/src/main/scala/za/co/absa/spline/core/harvester/componentCreators.scala

```diff
@@ -140,7 +142,11 @@ object ExpressionConverter {
     def render(o: Any): Option[Any] = renderValue(render)(o)
 
     lit.value match {
```

Comment:
Avoid `asInstanceOf`. What about doing unapply on a `Literal` instead of a literal value?
```suggestion
lit match {
   case Literal(ad: ArrayData, ArrayType(elementType, _)) => ...
   case Literal(v, _) => ...
}
```